src/emitter.js

```diff
@@ -0,0 +1,279 @@
+		/**
+		 * Registers a callback function to be executed when an event is fired in a specific (emitter) object.
+		 *
+		 * @param {Emitter} emitter The object that fires the event.
+		 * @param {String} event The name of the event.
+		 * @param {Function} callback The function to be called on event.
+		 * @param {Object} [ctx] The object that represents `this` in the callback. Defaults to `emitter`.
+		 * @param {Number} [priority] The priority of this callback in relation to other callbacks to that same event.
+		 * Lower values are called first. Defaults to `10`.
+		 */
+		listenTo: function( emitter, event, callback, ctx, priority ) {
+			var emitters, emitterId, emitterInfo, eventCallbacks;
+
+			// _listeningTo contains a list of emitters that this object is listening to.
+			// This list has the following format:
+			//
+			// _listeningTo: {
+			//     emitterId: {
+			//         emitter: emitter,
+			//         callbacks: {
+			//             event1: [ callback1, callback2, ... ]
+			//             ....
+			//         }
+			//     },
+			//     ...
+			// }
+
+			if ( !( emitters = this._listeningTo ) ) {
+				emitters = this._listeningTo = {};
+			}
+
+			if ( !( emitterId = emitter._emitterId ) ) {
+				emitterId = emitter._emitterId = utils.uid();
+			}
+
+			if ( !( emitterInfo = emitters[ emitterId ] ) ) {
+				emitterInfo = emitters[ emitterId ] = {
+					emitter: emitter,
+					callbacks: {}
+				};
+			}
+
+			if ( !( eventCallbacks = emitterInfo.callbacks[ event ] ) ) {
+				eventCallbacks = emitterInfo.callbacks[ event ] = [];
+			}
+
+			eventCallbacks.push( callback );
+
+			// Finally register the callback to the event.
+			emitter.on( event, callback, ctx, priority );
+		},
+
+		/**
+		 * Stops listening for events. It can be usued at different levels:
+		 *
+		 *  * To stop listening to a specific callback.
+		 *  * To stop listening to a specific event.
+		 *  * To stop listening to all events fired by a specific object.
+		 *  * To stop listening to all events fired by all object.
+		 *
+		 * @param {Emitter} [emitter] The object to stop listening to. If omitted, stops it for all objects.
+		 * @param {String} [event] (Requires `emitter`) The name of the event to stop listening to. If omitted, stops it
+		 * for all events from `emitter`.
+		 * @param {Function} callback (Requires `event`) The function be removed from the call list for the give
+		 * `event`.
+		 */
+		stopListening: function( emitter, event, callback ) {
+			var emitters = this._listeningTo;
+			var emitterId = emitter && emitter._emitterId;
+			var emitterInfo = emitters && emitterId && emitters[ emitterId ];
+			var eventCallbacks = emitterInfo && event && emitterInfo.callbacks[ event ];
+
+			// Stop if nothing has been listened.
+			if ( !emitters || ( emitter && !emitterInfo ) || ( event && !eventCallbacks ) ) {
+				return;
+			}
+
+			if ( callback ) {
+				// All params provided. off() that single callback.
+				emitter.off( event, callback );
```

Comment:
> And what those scenarios are?  

Just like a button click. You can do button.on( ‘click’ ) or me.listenTo( button, ‘click’ ). Then I want to listen to other stuff from button, but during the execution lifecycle I may decide to stop listening to ‘click’ only… and at the end of the execution I could stop listening to everything.

> TBH, that should not be an argument. This way we will accept everything if it's already coded and working.

In this case we’re talking about a pretty standard API that we “expect” to be used. We’re not inventing anything new here. That’s the way people are used to have such kind of API out there, where you pass the same arguments to listen and stop listening to events.

> And these series are IMO unnecessarily complex.

Well, they’re the very minimum needed to have those feature available.

Ofc if the above arguments don’t convince we can remove features and code (and risk to have to get then back later).

Still the idea of having a `{ off() }` object returned by `on()` sounds good and it definitely fits a new issue.